app/src/androidTest/java/org/simple/clinic/overdue/AppointmentRepositoryAndroidTest.kt

```diff
@@ -934,16 +1116,227 @@ class AppointmentRepositoryAndroidTest {
+
+    scheduleAppointment(appointmentUuidForFirstPatient, firstPatient).blockingGet()
+    scheduleAppointment(appointmentUuidForSecondPatient, secondPatient).blockingGet()
+
+    // when
+    val bloodSugarByAppointmentUuid = appointmentRepository
+        .overdueAppointments(since = LocalDate.now(clock), facility = facility)
+        .blockingFirst()
+        .associateBy({ it.appointment.uuid }, { it.patientLastSeen })
+
+    // then
+    val expected = mapOf(
+        appointmentUuidForFirstPatient to laterRecordedBloodSugarForFirstPatient.recordedAt,
+        appointmentUuidForSecondPatient to laterRecordedBloodSugarForSecondPatient.recordedAt
+    )
+    assertThat(bloodSugarByAppointmentUuid).isEqualTo(expected)
+  }
+
+  @Test
+  fun when_picking_overdue_appointment_and_blood_sugar_is_latest_compared_to_blood_pressure_then_blood_sugar_should_be_considered() {
+    fun createBloodSugar(patientProfile: PatientProfile, recordedAt: Instant): BloodSugarMeasurement {
+      return testData.bloodSugarMeasurement(
+          patientUuid = patientProfile.patient.uuid,
+          recordedAt = recordedAt
+      )
+    }
+
+    fun createBloodPressure(patientProfile: PatientProfile, recordedAt: Instant): BloodPressureMeasurement {
+      return testData.bloodPressureMeasurement(
+          patientUuid = patientProfile.patient.uuid,
+          recordedAt = recordedAt
+      )
+    }
+
+    fun scheduleAppointment(
+        appointmentUuid: UUID,
+        patientProfile: PatientProfile
+    ): Single<Appointment> {
+      return appointmentRepository.schedule(
+          patientUuid = patientProfile.patient.uuid,
+          appointmentUuid = appointmentUuid,
+          appointmentDate = LocalDate.parse("2017-12-30"),
+          appointmentType = Manual,
+          appointmentFacilityUuid = facility.uuid,
+          creationFacilityUuid = facility.uuid
+      )
+    }
+
+    // given
+    val firstPatient = testData.patientProfile(
+        patientUuid = UUID.fromString("e1943cfb-faf0-42c4-b5b6-14b5153295b2"),
+        generatePhoneNumber = true
+    )
+
+    patientRepository.save(listOf(firstPatient)).blockingAwait()
+
+    val earlierRecordedBPForFirstPatient = createBloodPressure(
+        patientProfile = firstPatient,
+        recordedAt = Instant.parse("2017-12-31T23:59:59Z")
+    )
+    val laterRecordedBloodSugarForFirstPatient = createBloodSugar(
+        patientProfile = firstPatient,
+        recordedAt = Instant.parse("2018-01-01T00:00:00Z")
+    )
+
+    bpRepository.save(listOf(earlierRecordedBPForFirstPatient))
+    bloodSugarRepository.save(listOf(laterRecordedBloodSugarForFirstPatient)).blockingAwait()
+
+    val appointmentUuidForFirstPatient = UUID.fromString("d9fd734d-13b8-43e3-a2d7-b40341699050")
+
+    scheduleAppointment(appointmentUuidForFirstPatient, firstPatient).blockingGet()
+    scheduleAppointment(appointmentUuidForFirstPatient, firstPatient).blockingGet()
```

Comment:
Since we have only one patient in this test, I think we can drop the "First" qualifiers from the variable names.